app/renderer.js

```diff
@@ -107,10 +107,27 @@ module.exports = class Renderer {
     // ## Public methods
     // Renders given file with given options
     renderFile(files, options, done) {
+        let promise;
+
         if (!done && typeof options === "function") {
             done = options;
             options = {};
         }
+
+        if (!done && typeof Promise === 'function') {
```

Comment:
yamp only supports node.js versions > 4.0.0 as specified in the _package.json_
```json
  "engines": {
    "node": ">= 4.0.0"
  }
```

I think Promises are completely supported in Node 4 so it shouldn't be necessary that check